### PR TITLE
[BUGFIX] Add connect args to execution engine schema

### PR DESCRIPTION
**Pull request by itaise**
This PR is to solve [this issue](https://github.com/great-expectations/great_expectations/issues/6226).

There is an issue that when creating SQLAlchemy execution engine, if the user send "connect_args" in the execution engine config, it is not passed to the "create_engine" method. 
This is because this attribute is dropped when creating the ExecutionEngineConfigSchema.

I checked it locally and this PR solve the issue

Changes proposed in this pull request:
- Add connect_args to the ExecutionEngineConfigSchema

### CODE REVIEW

**Review by amitbend — APPROVED** [review]

File: great_expectations/data_context/types/base.py [hunk 1]
```diff
@@ -1004,6 +1004,9 @@ class Meta:
     boto3_options = fields.Dict(
         keys=fields.Str(), values=fields.Str(), required=False, allow_none=True
     )
+    connect_args = fields.Dict(
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/types/base.py:1010]: Would we be able to add `connect_args` to `ExecutionEngineConfig` class above as well?

Comment (Shinnnyshinshin) [reply great_expectations/data_context/types/base.py:1010]: Otherwise the PR looks great. thanks for catching this @itaise 🙇🏼

Comment (itaise) [reply great_expectations/data_context/types/base.py:1010]: @Shinnnyshinshin  thank you! done :)

**Review by Shinnnyshinshin — APPROVED** [review]
🚀

**Pull request merged**